### PR TITLE
Add search in solutions

### PR DESCRIPTION
**Pull request by zaelgohary**
### Description

- Add search in Application page

### Changes

- Filter applications based on the search item

### Related Issues

- https://github.com/threefoldtech/tfgrid-sdk-ts/issues/2117

### Checklist

- [ ] Tests included
- [ ] Build pass
- [ ] Documentation
- [ ] Code format and docstrings
- [ ] Screenshots/Video attached (needed for UI changes)

### CODE REVIEW

**Comment by AlaaElattar** [comment]
- when i searched then clicked, it didn't open directly.

[Screencast from 02-18-2024 02:16:02 PM.webm](https://github.com/threefoldtech/tfgrid-sdk-ts/assets/54944491/27af54da-02d1-472e-9c99-fb833b2ba47b)

**Comment by MohamedElmdary** [comment]
> * when i searched then clicked, it didn't open directly.
> 
>  Screencast.from.02-18-2024.02.16.02.PM.webm

I don't think we can use combox because it's default searching will be remove after any `click` event no matter we stop propagation or not because it's not about `click` but `blur` event

So I'd recommend using `v-text-frield` instead as we don't need to select from list but search with some text

File: packages/playground/src/views/solutions_view.vue [hunk 1]
```diff
@@ -1,12 +1,6 @@
 <template>
   <view-layout>
-    <v-autocomplete
-      label="Search Applications"
-      v-model="searchItem"
-      :items="titles"
-      class="mb-5"
-      @update:search="filterCards"
-    ></v-autocomplete>
+    <v-text-field label="Search Applications" v-model="searchItem" class="mb-5" @input="filterCards"></v-text-field>
```
Comment (MohamedElmdary) [inline packages/playground/src/views/solutions_view.vue:3]: I think it should be clearable

Comment (zaelgohary) [reply packages/playground/src/views/solutions_view.vue:3]: what do you mean?

Comment (MohamedElmdary) [reply packages/playground/src/views/solutions_view.vue:3]: clearable prop in vuetify `v-text-field`

**Review by AlaaElattar — APPROVED** [review]

File: packages/playground/src/views/solutions_view.vue (hunk 1 above)
Comment (zaelgohary) [reply packages/playground/src/views/solutions_view.vue]: Added.

File: packages/playground/src/views/solutions_view.vue [hunk 2]
```diff
@@ -1,7 +1,14 @@
...
+    <v-text-field
+      label="Search Applications"
+      v-model="searchItem"
+      class="mb-5"
+      @input="filterCards"
+      clearable
+    ></v-text-field>
     <v-row>
-      <v-col sm="12" md="6" lg="4" v-for="card in cards" :key="card.title">
+      <v-col sm="12" md="6" lg="4" v-for="card in filteredCards" :key="card.title">
```
Comment (MohamedElmdary) [inline packages/playground/src/views/solutions_view.vue:11]: here we can update filter directly so we can eliminate the entire logic

File: packages/playground/src/views/solutions_view.vue [hunk 3]
```diff
@@ -161,11 +170,25 @@ export default {
         route: DashboardRoutes.Applications.Wordpress,
       },
     ];
+    const filteredCards = ref(cards);
+    const titles = cards.map(({ title }) => ({
+      title,
+    }));
```
Comment (MohamedElmdary) [inline packages/playground/src/views/solutions_view.vue:176]: If we applied the above suggestion we can remove this code
```suggestion
```

File: packages/playground/src/views/solutions_view.vue [hunk 4]
```diff
@@ -161,11 +170,25 @@ export default {
...
     const baseURL = import.meta.env.BASE_URL;
+    const searchItem = ref("");
+    function filterCards() {
+      filteredCards.value = cards.filter(n =>
+        n.title.toLocaleLowerCase().includes(searchItem.value.toLocaleLowerCase()),
+      );
+    }
```
Comment (MohamedElmdary) [inline packages/playground/src/views/solutions_view.vue:183]: If we applied the above suggestion we can remove this code
```suggestion
```

**Comment by MohamedElmdary** [comment]
Also if a user wrote a search query which doesn't  match any application we should show a message with some text message (e.g `No solution was found with the provided search query`)

File: packages/playground/src/views/solutions_view.vue [hunk 5]
```diff
@@ -1,7 +1,14 @@
 <template>
   <view-layout>
+    <v-text-field
+      label="Search Applications"
+      v-model="searchItem"
+      class="mb-5"
+      @input="filterCards"
+      clearable
+    ></v-text-field>
```
[the remark this answers is not in the document]
Comment (zaelgohary) [reply packages/playground/src/views/solutions_view.vue:9]: Replaced function logic w computed variable.

**Review by MohamedElmdary — APPROVED** [review]

**Pull request merged**